### PR TITLE
Implement nanmax function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
Adds keras.ops.nanmax, which computes the maximum of elements along a specified axis while ignoring NaN values in the input tensor.
The operation follows NumPy semantics for axis and keepdims and is supported across NumPy, TensorFlow, PyTorch, and JAX backends. Not supported on OpenVINO.

### CODE REVIEW

File: keras/src/ops/numpy.py [hunk 1]
```diff
@@ -5064,6 +5064,62 @@ def moveaxis(x, source, destination):
...
+@keras_export(["keras.ops.nanmax", "keras.ops.numpy.nanmax"])
+def nanmax(x, axis=None, keepdims=False):
+    """Maximum of a tensor over the given axes, ignoring NaNs.
+
+    Args:
+        x: Input tensor.
+        axis: Axis or axes along which the maximum is computed.
+            The default is to compute the maximum of the flattened tensor.
+        keepdims: If this is set to `True`, the axes which are reduced are left
+            in the result as dimensions with size one.
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/ops/numpy.py]: I've added default value for the description.
